src/sentry/api/base.py

```diff
@@ -83,6 +84,12 @@
 audit_logger = logging.getLogger("sentry.audit.api")
 api_access_logger = logging.getLogger("sentry.access.api")
 
+DEFAULT_SLUG_PATTERN = r"^(?![0-9]+$)[a-z0-9_\-]+$"
```

Comment:
`(?![0-9]+$)` -  negative lookahead to prevent only numbers
`[a-z0-9_\-]` - rest of the slug is made of alphanumeric characters, hyphens, and underscores